apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/AuthenticationProviderActivity.java

```diff
@@ -93,9 +114,75 @@ public void onClick(View v) {
                 }
             }
         }));
-        ListView listView = findViewById(R.id.list);
-        listView.setAdapter(new TestFeaturesListAdapter(featureList));
-        listView.setOnItemClickListener(TestFeatures.getOnItemClickListener());
+        mListView = findViewById(R.id.list);
+        loadAuthStatus(sUserInformation == null);
+        mListView.setOnItemClickListener(TestFeatures.getOnItemClickListener());
+    }
+
+    private static boolean isAuthenticated() {
+        if (sUserInformation == null) {
+            return false;
+        }
+
+        /* TODO Call method directly / remove reflection once SDK being released. */
+        boolean accessTokenIsNull = false;
+        try {
+            Method method = UserInformation.class.getMethod("getAccessToken");
+            Object accessToken = method.invoke(sUserInformation);
+            accessTokenIsNull = accessToken == null;
+        } catch (NoSuchMethodException ignore) {
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        return !accessTokenIsNull;
+    }
+
+    private void loadAuthStatus(boolean _default) {
```

Comment:
We do not use underscores in variable names, please instead add a suffix defaultSomething